sdk/test/logs/logger_provider_sdk_test.cc

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/logs/provider.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/sdk/logs/logger_provider.h"
+
+#include <gtest/gtest.h>
+
+using namespace opentelemetry::sdk::logs;
+
+TEST(LoggerProviderSDK, PushToAPI)
+{
+  auto lp = opentelemetry::nostd::shared_ptr<opentelemetry::logs::LoggerProvider>(
+      new opentelemetry::sdk::logs::LoggerProvider());
+  opentelemetry::logs::Provider::SetLoggerProvider(lp);
+
+  // Check that the loggerprovider was correctly pushed into the API
+  ASSERT_EQ(lp, opentelemetry::logs::Provider::GetLoggerProvider());
+}
+
+TEST(LoggerProviderSDK, LoggerProviderGetLogger)
+{
+  LoggerProvider lp;
+
+  auto logger1 = lp.GetLogger("logger1");
+  auto logger2 = lp.GetLogger("logger2");
+
+  // Check that the logger is not nullptr
+  ASSERT_NE(logger1, nullptr);
+
```

Comment:
Please add test case what happens if we try calling Logger::log() here ( i.e, no processor initialized ).